CedarJavaFFI/src/interface.rs

```diff
@@ -49,25 +59,25 @@ fn call_cedar_in_thread(call_str: String, input_str: String) -> String {
 /// The main JNI entry point
 #[jni_fn("com.cedarpolicy.BasicAuthorizationEngine")]
 pub fn callCedarJNI(
-    env: JNIEnv<'_>,
+    mut env: JNIEnv<'_>,
     _class: JClass<'_>,
     j_call: JString<'_>,
     j_input: JString<'_>,
 ) -> jstring {
-    let parsing_err = || build_err_obj(env, "parsing");
-    let getting_err = || build_err_obj(env, "getting");
+    let parsing_err = build_err_obj(&env, "parsing");
+    let getting_err = build_err_obj(&env, "getting");
 
-    let j_call_str: String = match env.get_string(j_call) {
+    let j_call_str: String = match env.get_string(&j_call) {
         Ok(call_str) => call_str.into(),
-        _ => return getting_err(),
+        _ => return getting_err,
```

Comment:
would it be better to just inline the call to `build_err_obj()` here? so we don't construct the error object in Java unless we're actually returning the error?